### PR TITLE
Fix some part of the modules.

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
The script where this raised an error was 
```julia
IP1 = projective_space(QQ, [:s, :t])
B = covered_scheme(IP1)
E = direct_sum([twisting_sheaf(IP1, k) for k in [-4, -6, 1]])
IPE = projectivization(E, var_names=["z", "x", "y"])
X = covered_scheme(IPE)
pr = covered_projection_to_base(IPE)
U = first(affine_charts(X))
(x, y, t) = gens(OO(U))
f = y^2 -(x^3 + t^5*(t-1)^2)
I = IdealSheaf(X, U, [f])
inc = Oscar.CoveredClosedEmbedding(X, I)
S = domain(inc)
I_sing_S = Oscar.ideal_sheaf_of_singular_locus(S)
I_sing_X = radical(pushforward(inc, I_sing_S))
singular_components = Oscar.maximal_associated_points(I_sing_X)
Y = X
iota = inc
projections = AbsCoveredSchemeMorphism[]
blowdowns = BlowupMorphism[]
k = 0 # A variable counting the blowups
  global k = k + 1
  global iota
  global projections
  J = radical(first(singular_components))
  @show "blowing up $J"
  for U in affine_charts(scheme(J))
    @show gens(J(U))
  end
  cov = Oscar.simplified_covering(Y)
  @assert cov in coverings(Y)
  global bl = blow_up(J, covering=cov, var_name=Symbol("u$(k)_"))
  push!(blowdowns, bl)
  Sup, iota, pr = strict_transform(bl, iota)
  push!(projections, pr)
  global I_sing_S = Oscar.ideal_sheaf_of_singular_locus(Sup)
  @show isone(I_sing_S)
  @show dim(I_sing_S)
  global I_sing_X = radical(pushforward(iota, I_sing_S))
  @show isone(I_sing_X)
  @show I_sing_X
  @show dim(I_sing_X)
  for U in affine_charts(scheme(I_sing_X))
    @show gens(I_sing_X(U))
  end
  global singular_components = Oscar.maximal_associated_points(I_sing_X)
```
which is certainly not a minimal example. But it runs now. 

CC: @jankoboehm 

Edit: Actually, when executed in the REPL, the above code also produces another error due to printing. Ignore that one, please. It will be fixed elsewhere.

### CODE REVIEW

File: src/Modules/UngradedModules.jl [hunk 1]
```diff
@@ -1194,6 +1194,15 @@ function SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}
   return subModule
 end
 
+# missing method
+function sparse_row(A::MatElem)
```
Comment (HechtiDerLachs) [inline src/Modules/UngradedModules.jl:1197]: This is called in the method just above this block. That's why I put it here for the time being.

File: src/Modules/UngradedModules.jl [hunk 2]
```diff
@@ -1194,6 +1194,15 @@ function SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}
   return subModule
 end
 
+# missing method
```
Comment (simonbrandhorst) [inline src/Modules/UngradedModules.jl]:
```suggestion
```
Not missing anymore.

**Comment by simonbrandhorst** [comment]
Would you add a test for `SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L} `  which calls `sparse_row`?
Then I think this is good to go.

**Comment by lgoettgens** [comment]
The added function was moved to Hecke last week in https://github.com/thofma/Hecke.jl/pull/1221 and https://github.com/oscar-system/Oscar.jl/pull/2846.
Please double-check that the error still occurs when you work in the most-recent Oscar master and after running `]up`.
If it is still there, could you provide the error message here? I would really like to understand what went wrong with the two PRs.

**Comment by lgoettgens** [comment]
It could just be a problem of just a missing import or so.

**Comment by lgoettgens** [comment]
I just ran your above code against Hecke 0.12.1 and it worked. So I see no point in this change.

File: src/Modules/UngradedModules.jl [hunk 3]
```diff
@@ -1194,6 +1194,14 @@ function SubModuleOfFreeModule(F::FreeMod{L}, A::MatElem{L}) where {L}
   return subModule
 end
 
+function sparse_row(A::MatElem)
```
Comment (fingolfin) [inline src/Modules/UngradedModules.jl:1197]: Didn't we just remove this? Have you updated OSCAR master and then run `using Pkg ; Pkg.update()` to make sure you get the latest Hecke, Nemo etc.?

Comment (fingolfin) [reply src/Modules/UngradedModules.jl:1197]: Sorry, I had this open in a tab from last night and did not see the comments which already say this. But I can confirm: your code works fine for me with latest Oscar master and all packages up-to-date, except for the printing error.

**Pull request closed without merging**